src/commands/commerce/extension/unmap.ts

```diff
@@ -0,0 +1,116 @@
+            description: msgs.getMessage('extension.unmap.regExtensionNameFlagDescription'),
+        }),
+        'store-name': flags.string({
+            char: 'n',
+            description: msgs.getMessage('extension.unmap.StoreNameFlagDescription'),
+        }),
+        'store-id': flags.string({
+            char: 'i',
+            description: msgs.getMessage('extension.unmap.storeId'),
+        }),
+    };
+    public org: Org;
+    public statusFileManager: StatusFileManager;
+
+    // eslint-disable-next-line @typescript-eslint/require-await
+    public async run(): Promise<void> {
+        this.unmapRecord(
+            this.flags['registered-extension-name'],
+            this.flags['store-name'],
+            this.flags['store-id'],
+            this.org.getUsername()
+        );
+    }
+
+    public unmapRecord(extensionName: string, storeName: string, storeId: string, userName: string): void {
+        if (storeName === undefined && storeId === undefined) {
+            throw new SfdxError(msgs.getMessage('extension.unmap.undefinedName'));
+        }
+        const storeid = this.validateStoreId(storeName, storeId, userName);
+        const name = forceDataSoql(`SELECT Name FROM WebStore WHERE Id='${storeid}' LIMIT 1`).result.records[0].Name;
+        try {
+            let deletedId: string;
+            const existingIds = forceDataSoql(
+                `SELECT DeveloperName,ExternalServiceProviderType FROM RegisteredExternalService WHERE DeveloperName='${extensionName}' AND ExternalServiceProviderType='Extension'`,
+                userName
+            );
+            for (const record of existingIds.result.records) {
+                const id = (record['ExternalServiceProviderType'] as string)
+                    .concat('__' as string)
+                    .concat(record['DeveloperName'] as string);
+                const deleteId = forceDataSoql(
+                    `SELECT Id FROM StoreIntegratedService WHERE Integration='${id}' AND StoreId='${storeid}'`,
+                    userName
+                ).result.records[0];
+                deletedId = deleteId.Id;
+                forceDataRecordDelete('StoreIntegratedService', deletedId, this.org.getUsername(), 'pipe');
+                this.ux.log(
+                    msgs.getMessage('extension.unmap.savingConfigIntoConfig', [
+                        // eslint-disable-next-line @typescript-eslint/restrict-template-expressions
+                        `'${record['DeveloperName']}'`,
+                        'from your webstore',
+                        `'${name}'`,
+                    ])
+                );
+            }
+        } catch (e) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+            throw new SfdxError(msgs.getMessage('extension.unmap.error', [extensionName, '\n', e.message]));
+        }
+    }
+
+    private validateStoreId(storeName: string, storeId: string, userName: string): string {
```

Comment:
this function is a copy and paste of the one in map.ts would we consider moving this to a util class which both unmap and map can reference